### ISSUE TITLE
Async Await Example in Transactions

### ISSUE THREAD

**Issue by adarshmadrecha** [knex/knex#5879]
Dear Contributors, It would be amazing to have `Async-Await` example in the [transaction](https://knexjs.org/#Transactions) section 

Example Code: https://github.com/tgriesser/knex/issues/3309#issuecomment-509613834

This will also clear out the confusion on what is the best way to write `Async-Await` style code when using transaction. https://github.com/tgriesser/knex/issues/3309 also has some discussion on this.

**Comment by mercmobily** [comment]
You're absolutely right that the transaction documentation would benefit from async/await examples. The current documentation primarily shows callback-based patterns and promise chains, which aren't how most modern Node.js code is written these days.

Looking at the current documentation, it does now include some TypeScript examples using async/await (such as in the transaction modes section and the transactionProvider examples), but the main transaction section still relies heavily on promise chains with `.then()` and `.catch()`.

**What would be helpful:**

Adding clear async/await examples for common transaction patterns:

```js
// Using trx as a query builder
try {
  await knex.transaction(async (trx) => {
    const ids = await trx('catalogues').insert({name: 'Canterbury Tales'});
    await trx('books').insert({catalogue_id: ids[0], title: 'The Tale'});
  });
  console.log('Transaction complete');
} catch (error) {
  console.log('Transaction rolled back:', error);
}
```

```js
// Manual transaction control
const trx = await knex.transaction();
try {
  await trx('users').insert({name: 'John'});
  await trx('logs').insert({action: 'user_created'});
  await trx.commit();
} catch (error) {
  await trx.rollback();
  throw error;
}
```

As elhigu noted, using explicit commit/rollback (as shown in the linked comment #3309) is generally not the best approach. The preferred pattern is to use the transaction callback function, which automatically commits on success and rolls back on error. However, showing both patterns with async/await syntax would still be valuable.


**Related Issues:**
- #3309 - Discussion about async/await with transactions
- #5887 - Need to document exact behaviour of trx.commit() and trx.rollback()
- #5893 - Incomplete transactionProvider example
